stixcore/soc/manager.py

```diff
@@ -0,0 +1,86 @@
+import os
+from pathlib import Path
+from binascii import unhexlify
+from xml.etree import ElementTree as Et
+from collections import defaultdict
+
+from stixcore.tmtc.packets import TMTC, TMPacket
+from stixcore.util.logging import get_logger
+
+__all__ = ['SOCManager', 'SOCPacketFile']
+
+
+logger = get_logger(__name__)
+
+
+class SOCPacketFile:
+    def __init__(self, file):
+        self.file = Path(file)
+        if not self.file.exists():
+            raise ValueError(f"file path not found: {file}")
+        stat = self.file.stat()
+        self.size = stat.st_size
+        self.file_date = stat.st_ctime
+        self.spacecraft_date = None  # TODO some file name conventions?
+        self.tmtc = TMTC.All
+        self.packet_data = defaultdict(list)
+
+        with open(self.file, "r") as f:
+            header = f.read(1000)
+
+            if 'Response' not in header:
+                raise ValueError("xml does not contain any 'Response'")
+            if 'PktRawResponse' in header:
+                self.tmtc = TMTC.TM
+            elif 'PktTcReportResponse' in header:
+                self.tmtc = TMTC.TC
+            else:
+                raise ValueError("xml does not contain any TM or TC response")
+
+    def read_packet_data_header(self):
+        root = Et.parse(str(self.file)).getroot()
+        if self.tmtc == TMTC.TC:
```

Comment:
I would hope both these beaches can be combined using TMTCPacket() but might take some work.